### PR TITLE
cloud_functions: add getReobserveVaas

### PR DESCRIPTION
**Pull request by panoel**
This PR add a cloud function that will return VAAs that may need to be reobserved.

### CODE REVIEW

File: cloud_functions/src/alarmMissingVaas.ts [hunk 1]
```diff
@@ -332,6 +378,24 @@ async function storeAlarmedChains(alarms: AlarmedChainTime[]): Promise<void> {
...
+    // curl -X 'GET' \
+    // 'https://api.wormholescan.io/v1/signed_vaa/1/ec7372995d5cc8732397fb0ad35c0121e0eaa90d26f828a534cab54391b3a4f5/319118' \
+    // -H 'accept: application/json'
+    // This function will return true if the get returns 200
+    // Otherwise, it will return false
+    if (response.status === 200) {
+      return true;
+    }
+  } catch (e) {
+    return false;
```
Comment (barnjamin) [inline cloud_functions/src/alarmMissingVaas.ts:394]: console.error?

File: cloud_functions/src/getReobserveVaas.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    // curl -X 'GET' \
+    // 'https://api.wormholescan.io/v1/signed_vaa/1/ec7372995d5cc8732397fb0ad35c0121e0eaa90d26f828a534cab54391b3a4f5/319118' \
+    // -H 'accept: application/json'
+    // This function will return true if the get returns 200
+    // Otherwise, it will return false
+    if (response.status === 200) {
+      return true;
+    }
+  } catch (e) {
+    return false;
```
Comment (barnjamin) [inline cloud_functions/src/getReobserveVaas.ts:91]: console.error?

File: cloud_functions/src/alarmMissingVaas.ts [hunk 3]
```diff
@@ -114,7 +126,11 @@ export async function alarmMissingVaas(req: any, res: any) {
     console.log('could not get missing VAAs', e);
     res.sendStatus(500);
   }
-  await updateFirestore(firestoreVAAs);
+  let reobs: ReobserveInfo[] = [];
+  reobsMap.forEach((vaa) => {
+    reobs.push(vaa);
+  });
+  await updateFirestore(firestoreVAAs, reobs);
```
Comment (barnjamin) [inline cloud_functions/src/alarmMissingVaas.ts:133]:
```suggestion
  await updateFirestore(firestoreVAAs, reobsMap.values());
```

File: cloud_functions/src/alarmMissingVaas.ts [hunk 4]
```diff
@@ -199,13 +215,43 @@ async function getAndProcessFirestore(): Promise<Map<string, FirestoreVAA>> {
...
+          const vaas: ReobserveInfo[] = data.VAAs;
+          vaas.forEach((vaa) => {
+            current.set(vaa.txhash, vaa);
+          });
+          console.log('number of reobserved VAAs', vaas.length);
+        }
+      }
+    })
+    .catch((error) => {
+      console.log('Error getting Reobserve document:', error);
```
Comment (barnjamin) [inline cloud_functions/src/alarmMissingVaas.ts:241]: .error?

File: cloud_functions/src/alarmMissingVaas.ts (hunk 1 above)
Comment (panoel) [reply cloud_functions/src/alarmMissingVaas.ts:394]: Fixed

File: cloud_functions/src/alarmMissingVaas.ts (hunk 3 above)
Comment (panoel) [reply cloud_functions/src/alarmMissingVaas.ts:133]: Does not appear to work without converting to unknown and then to ReobserveInfo[], which I would rather not do.

File: cloud_functions/src/getReobserveVaas.ts (hunk 2 above)
Comment (panoel) [reply cloud_functions/src/getReobserveVaas.ts]: Function has been moved to utils.ts

**Review by barnjamin — APPROVED** [review]

**Pull request merged**